### PR TITLE
Update to -std=c++14, remove custom make_unique

### PR DESCRIPTION
**Pull request by drzaeus77**
This updates our compiler requirements to approximately gcc 4.9/5.0 and
clang 3.4.

Signed-off-by: Brenden Blanco <bblanco@gmail.com>

### CODE REVIEW

File: src/cc/common.h [hunk 1]
```diff
@@ -23,11 +23,7 @@
 
 namespace ebpf {
 
-template <class T, class... Args>
-typename std::enable_if<!std::is_array<T>::value, std::unique_ptr<T>>::type
-make_unique(Args &&... args) {
-  return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
-}
+using std::make_unique;
```
Comment (goldshtn) [inline src/cc/common.h]: Do we really need this in the header? Why?

Comment (drzaeus77) [inline src/cc/common.h]: Less code churn. I don't mind tracking down the few uses and fixing those up.

**Comment by 4ast** [comment]
could you define precisely what is the minimum version of clang and gcc that would be required?
i'm worried about breaking distro builds. I think some are still running 4.8.
What is the advantage? other than it looks cleaner.

**Comment by drzaeus77** [comment]
For clang it is 3.4 minimum required. For gcc it is 4.9.

C++14 brings mostly ergonomic improvements. For instance, see the automatic default member initializer section in http://en.cppreference.com/w/cpp/language/data_members#Member_initialization, which I make use of in the new definition of the TableDesc and FileDesc classes in https://github.com/iovisor/bcc/pull/1040/files#diff-518cfb5455fc552447ecf859f684527dR92.

**Comment by drzaeus77** [comment]
> i'm worried about breaking distro builds. 

Is centos the weak link here?

**Comment by 4ast** [comment]
centos7 is still on gcc4.8 and I believe more or less recent fedora as well.
can we ask distro folks ?

**Comment by drzaeus77** [comment]
Fedora 21 was the first to support gcc 4.9...not that recent. I'm pretty sure if centos7 hasn't updated by now then they never will. Ubuntu 14.04 also is stuck on 4.8. But we're not building packages for either of those, so I don't have any visibility into what would break, and who is tracking this. I guess based on the other issue that was getting comments today, debian folks are interested as well.

**Comment by 4ast** [comment]
imo it feels to early to dump gcc4.8
@brendangregg @goldshtn thoughts?

**Comment by goldshtn** [comment]
I feel that C++ 14 is not worth breaking older distros. The cleanup introduced here is nice, but can easily be worked around with C++ 11 features.

**Pull request closed without merging**